src/utilities/coerceValue.js

```diff
@@ -46,7 +45,7 @@ export function coerceValue(
 ): CoercedValue {
   // A value must be provided if the type is non-null.
   if (isNonNullType(type)) {
-    if (isNullish(value)) {
+    if (value == null) {
```

Comment:
`== null` block `null` and `undefined` but not a NaN and simular values

https://github.com/graphql/graphql-js/blob/85971a277c4e7782e9be38d102bd6562dedf7804/src/jsutils/isNullish.js#L13-L15